### PR TITLE
Public API Coverage [API-1402]

### PR DESCRIPTION
**Pull request by emreyigit**
I have filled some of the missing parts. Coverage is in good shape. Missing parts are mostly get/set things or implicit invokes. 

And, the list is like (They will be handled in a sepereted task)
```
HazelcastOptionsBuilderBase.cs
HazelcastOptions.ClusterOptions.cs
HazelcastClientFactory.cs
MemberInfo.cs
HBigDecimal.cs
HLocalTime.cs
HLocalDate.cs
HOffsetDateTime.cs
HLocalDateTime.cs
MemberVersion.cs
MemberInfoState.cs
TargetUnreachableException.cs
LockOwnershipLostException.cs
LockAcquireLimitReachedException.cs
ClientNotAllowedInClusterException.cs
SqlServiceExtensions.cs
SqlRow.cs
SqlRowMetadata.cs
SqlColumnMetadata.cs
SqlStatementOptions.cs
HazelcastSqlException.cs
NearCacheOptions.cs
FactoryOptions.cs
SerializerOptions.cs
SerializationException.cs
InvalidPortableFieldException.cs
AuthenticationOptions.cs
MapEntryEventArgsBase.cs
MapEntryUpdatedEventArgs.cs
IndexConfigExtensions.cs
MapEntryAddedEventArgs.cs
MapEntryEvictedEventArgs.cs
MapEntryExpiredEventArgs.cs
TopicMessageEventHandler.cs
CollectionItemEventArgs.cs
MapEventArgsBase.cs
TopicMessageEventArgs.cs
TopicEventHandlers.cs
CollectionItemEventHandlers.cs
MapEventHandlers.cs
ReplicatedMapEventHandlers.cs
MultiMapEventHandlers.cs
MapEntryLoadedEventArgs.cs
MapEntryMergedEventHandler.cs
MapEntryInvalidatedEventArgs.cs
MapEntryInvalidatedEventHandler.cs

ConfigurationBuilderExtensions.cs
SslOptions.cs
ConnectionClosedEventArgs.cs
ConnectionOpenedEventArgs.cs
```

### CODE REVIEW

**Comment by srknzl** [comment]
Is this ready to review, if so, can you mark this ready for review? (remove draft status)

**Comment by zpqrtbnk** [comment]
Is this, like, raising code coverage from X% to Y%? Could you give the numbers, just to have an idea. Otherwise, looks good.

**Comment by emreyigit** [comment]
According to coverage report at GitHub actions, the coverage increased at least %1. I cannot compare fully right now since the PR is not seen at Codecov. After merged, I can provide more precise numbers.

**Review by zpqrtbnk — DISMISSED** [review]
Approved.

**Comment by srknzl** [comment]
@emreyigit can you somehow make sure public API coverage is 100% now? I ask because jira issue mentions 100%.

You said: 
> List of classes that needs more tests. Also, coverage looks in good shape. Missing parts are mostly get/set things or implicit invokes.

Won't you test getter and setters? I think they can be tested in other tests by using them in assertions etc. What do you mean by implicit invokes?

File: src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs [hunk 1]
```diff
@@ -489,6 +590,11 @@ public void Dispose()
             { }
         }
 
+        public class DummyBinder
+        {
+            public int Count { get; set; }
+        }
+
```
Comment (srknzl) [inline src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs:597]: Is this used somewhere?

**Comment by emreyigit** [comment]
@srknzl  this issue meant to be determine missing tests that decreases the public coverage. I just added few obvious tests while investigating. Real implementation will be handled in another issue. Also, in C#, we use properties which have built in getter and setters. They don't require any test.

**Comment by srknzl** [comment]
Yeah I agree getter/setters that does not have complex logic should not be tested. After you remove DummyBinder I will approve.

What do you mean by `Real implementation will be handled in another issue.`? By implementation I think you mean tests. So, more tests will be added later.

File: src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs (hunk 1 above)
Comment (emreyigit) [reply src/Hazelcast.Net.Tests/Configuration/HazelcastOptionsTests.cs]: Fixed at https://github.com/hazelcast/hazelcast-csharp-client/pull/679/commits/b18db521cd42644ac6a615e314f9b18525b6bfa5

**Review by srknzl — APPROVED** [review]

**Review by zpqrtbnk — APPROVED** [review]
Re-Approved

**Review by zpqrtbnk — APPROVED** [review]
Approve.

**Pull request merged**